WebApp/src/uk/ac/exeter/QuinCe/data/Dataset/QC/Routines/GradientTestRoutine.java

```diff
@@ -0,0 +1,124 @@
+package uk.ac.exeter.QuinCe.data.Dataset.QC.Routines;
+
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import uk.ac.exeter.QuinCe.data.Dataset.SensorValue;
+import uk.ac.exeter.QuinCe.data.Dataset.QC.Flag;
+
+public class GradientTestRoutine extends Routine {
+
+  /**
+   * The maximum delta between values, in units per minute
+   */
+  private double maxDelta;
+
+  /**
+   * Basic constructor
+   *
+   * @param parameters
+   *          The parameters
+   * @throws QCRoutinesConfigurationException
+   *           If the parameters are invalid
+   */
+  public GradientTestRoutine(List<String> parameters) throws RoutineException {
+    super(parameters);
+    // TODO Auto-generated constructor stub
```

Comment:
Get rid of this line. TODO is a special keyword for notes in code to remind us that things need doing at some point. In theory we should make issues for them too, but I often stick in a TODO while developing big things to remind me that I have to go back to them. And sometimes I forget, so I occasionally run a TODO search through all the code.